File: cs/Minsk/CodeAnalysis/Syntax/NameExpressionSyntax.cs
namespace Minsk.CodeAnalysis.Syntax;

public sealed class NameExpressionSyntax : ExpressionSyntax
{
    public NameExpressionSyntax(SyntaxToken identifierToken)
    {
        IdentifierToken = identifierToken;
    }

    public SyntaxToken IdentifierToken { get; }
    public override SyntaxKind Kind => SyntaxKind.NameExpression;
    public override IEnumerable<SyntaxNode> Children => new[] { IdentifierToken };
}

File: nim/src/minsk/codeAnalysis/syntax/expressions/nameExpressionSyntax.nim
import minsk/codeAnalysis/syntax/[
  expressionSyntax,
  syntaxKind,
  syntaxNode,
  syntaxToken,
]

type
  NameExpressionSyntax* = ref object of ExpressionSyntax
    identifierToken*: SyntaxToken

proc newNameExpressionSyntax*(identifierToken: SyntaxToken): NameExpressionSyntax =
  new(result)
  result.identifierToken = identifierToken

method kind*(self: NameExpressionSyntax): SyntaxKind =
  SyntaxKind.NameExpression

method children*(self: NameExpressionSyntax): seq[SyntaxNode] =
  result.add(self.identifierToken)
